pysradb/search.py

```diff
@@ -0,0 +1,1619 @@
+        case-sensitive exact matches for search queries.
+
+        Parameters
+        ----------
+        regex_matcher : dict
+            dictionary of regex expression : accepted string by API pairs.
+        input_query : str
+            input string for a particular query field.
+        error_message : str
+            error message to be shown if input_query does not match any of
+            the regex expressions in regex_matcher.
+
+        Returns
+        -------
+        tuple
+            tuple pair of (input_query, message).
+            message is "" if no format error has been identified, error
+            message otherwise.
+        """
+        matched_strings = []
+        for regex_expression in regex_matcher:
+            if re.match(regex_expression, input_query, re.IGNORECASE):
+                matched_strings.append(regex_matcher[regex_expression])
+        if not matched_strings:
+            return input_query, error_message
+        elif len(matched_strings) == 1:
+            return matched_strings[0], ""
+        else:
+            message = (
+                f"Multiple potential matches have been identified for {input_query}:\n"
+                f"{matched_strings}\n"
+                f"Please check your input.\n\n"
+            )
+            return input_query, message
+
+    def _validate_fields(self):
+        """Verifies that user input format is correct.
+
+        Raises
+        ------
+        IncorrectFieldException
+            If the input to any query field is in the wrong format
+
+        """
+
+        message = ""
+
+        # verify layout
+        if self.fields["layout"] and str(self.fields["layout"]).upper() not in [
+            "SINGLE",
+            "PAIRED",
+        ]:
+            message += (
+                f"Incorrect layout field format: {self.fields['layout']}\n"
+                "--layout must be either SINGLE or PAIRED\n\n"
+            )
+        # verify mbases
+        if self.fields["mbases"]:
+            try:
+                self.fields["mbases"] = int(self.fields["mbases"])
+                if self.fields["mbases"] <= 0:
+                    raise ValueError
+            except (ValueError, TypeError):
+                message += (
+                    f"Incorrect mbases format: {self.fields['mbases']}\n"
+                    f"--mbases must be a positive integer\n\n"
+                )
+        # verify publication_date
+        date_regex = "(0[1-9]|[12][0-9]|3[01])-(0[1-9]|1[012])-(19|20)[0-9]{2}"
+        if self.fields["publication_date"] and not re.match(
+            f"^{date_regex}(:{date_regex})?$", self.fields["publication_date"]
+        ):
+            message += (
+                f"Incorrect publication date format: {self.fields['publication_date']}\n"
+                f"Expected --publication-date format: dd-mm-yyyy or dd-mm-yyyy:dd-mm-yyyy, between 1900-2099\n\n"
+            )
+        # verify platform
+        platform_matcher = {
```

Comment:
Can you leave a comment here how these were determined? If a new platform is introduced in SRA/ENA how should this be changed?